.github/workflows/publish.yml

```
@@ -0,0 +1,53 @@
+name: "Publish NPM package to repository"
+
+on: 
+  workflow_dispatch:
+    inputs:
+      tag_version:
+        description: 'The tag_version, e.g. vx.x.x'
+        default: 'vx.x.x'
+        required: true
+        type: 'string'
+      node_version:
+        description: 'The node_version to build with.'
+        default: '22.x'
+        required: true
+        type: 'string'
+      dry_run:
+        description: 'Whether this is a dry_run publish.'
+        default: 'true'
+        required: true
+        type: 'boolean'
+
+jobs:
+  build_and_publish:
+    runs-on: ubuntu-latest
```

We should probably pin this to latest lts.

```suggestion
    runs-on: ubuntu-24.04
```